CONTRIBUTING.md

```diff
@@ -0,0 +1,158 @@
+# Contributing to Konflux UI
+
+Thanks for your interest in contributing to **Konflux UI**! 🎉  
+This document will help you get started with contributing code, documentation, or ideas.
+
+---
+
+## How to Contribute
+
+We welcome contributions of all kinds! Follow these steps to get started:
+
+1. **Fork the repository** and create your branch from `main`.
+
+2. Make sure your code builds and passes all checks:
+
+   - Lint (`yarn lint`)
+   - Type check (`yarn tsc`)
+   - Tests (`yarn test`)
+
+3. **Write or update tests**:
+
+   - Use [Jest](https://jestjs.io/) for unit tests.
+   - Add tests for new components, utility functions, or any logic added/changed.
+   - When fixing bugs or refactoring code, consider whether a new test would help prevent regressions.
+   - We use [Codecov](https://app.codecov.io/gh/konflux-ci/konflux-ui) to track test coverage and aim to maintain at least **80% coverage**.
+   - Pull Requests that significantly reduce coverage may be asked to add more tests.
+   - If your changes impact end-to-end behavior or UI flows, consider updating or adding to the e2e tests located in the `e2e-tests/` folder.  
+     For guidance on contributing to e2e tests, please refer to the [e2e-tests/README.md](e2e-tests/README.md) documentation.
+
+4. **Use feature flags when needed**:
+
+   - If you're contributing a feature that is not yet fully implemented or will be completed in follow-up PRs, please wrap it with a feature flag.
+   - Refer to our [feature flags guide](./docs/feature-flags.md) for more information.
+
+5. Submit a Pull Request following our [PR template](.github/PULL_REQUEST_TEMPLATE.md).  
+   The template includes required fields such as:
+
+   - **Fixes**: include references to related issues, if applicable
+   - **Description**: clear overview of what was changed
+   - **Screenshots**: required for any UI changes
+   - **How to test or reproduce**: how to verify the change
+
+6. **Tag reviewers in your Pull Request**:
```

Comment:
Would write somewhere to make sure to tag it as draft reviews are not required yet.